### PR TITLE
Create Dynamixel Class

### PR DESCRIPTION
**Pull request by maroqijalil**

### CODE REVIEW

File: src/tachimawari_dynamixel/dynamixel.cpp [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+
+#include <iostream>
+#include <iomanip>
+#include <memory>
+#include <string>
+#include <vector>
+
+using namespace std::chrono_literals;
+
+namespace tachimawari
```
Comment (threeal) [inline src/tachimawari_dynamixel/dynamixel.cpp]: Use `tachimawari_dynamixel` namespace instead of `tachimawari`.

File: include/tachimawari_dynamixel/dynamixel.hpp [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  explicit Dynamixel(
+    std::string node_name, std::string port = "tty/ACM0", float protocol_version = 0);
+  ~Dynamixel();
+
+  bool start();
+  void stop();
+
+private:
+  bool torque_enable(Joint joint);
+  bool torque_enable(std::vector<Joint> joints);
```
Comment (threeal) [inline include/tachimawari_dynamixel/dynamixel.hpp]: Use const reference for function parameters. Do this to the other functions too.

File: src/tachimawari_dynamixel/dynamixel.cpp [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+bool Dynamixel::init_joints_present_position(std::vector<Joint> joints)
+{
+  bool init_state = true;
+
+  if (!init_joints_state) {
+    init_state = sync_read_joints(joints);
+    this->joints = joints;
+    init_joints_state = true;
+  } else {
+    for (int index = 0; index < static_cast<int>(joints.size()); index++) {
```
Comment (threeal) [inline src/tachimawari_dynamixel/dynamixel.cpp]: Use `size_t` data type and remove `static_cast`. Do this to the other for loop too.

**Review by threeal — CHANGES_REQUESTED** [review]
See individual comment for the requested changes.

File: .github/workflows/build-and-test.yml [hunk 4]
```diff
@@ -1,17 +1,24 @@
...
+name: Build and Test Nightly
 on:
+  workflow_dispatch:
   pull_request:
     branches: [master]
   push:
     branches: [master]
 jobs:
-  build-and-test:
+  build-and-test-nightly:
```
Comment (threeal) [inline .github/workflows/build-and-test.yml]: Cither change the filename into `build-and-test-nightly.yml` or rename this into `build-and-test`.

File: include/tachimawari_dynamixel/dynamixel.hpp [hunk 5]
```diff
@@ -123,9 +125,9 @@ class Dynamixel : public rclcpp::Node
     int data_length = 4);
 
   bool move_joint(Joint joint);
-  bool move_joints(std::vector<Joint>);
+  bool move_joints(std::vector<Joint> & joints);
```
Comment (threeal) [inline include/tachimawari_dynamixel/dynamixel.hpp]: Actually it's not a good behavior to pass a parameter by reference and changes it inside the function, better to use a `shared_ptr`. But it's okay, it could be solved on another PR.

**Review by threeal — CHANGES_REQUESTED** [review]
Just a minor request on the workflow.

**Review by threeal — APPROVED** [review]
Looks good to me

**Review by finesaaa — APPROVED** [review]
Good 💯
